### PR TITLE
Removed deprecations in DependencyInjection component

### PR DESCRIPTION
**Pull request by dosten**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by dosten** [comment]
The tests didn't run, @fabpot can you run them manually?

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -681,72 +672,11 @@ private function addServices()
             } else {
                 $privateServices .= $this->addService($id, $definition);
             }
-
-            $synchronizers .= $this->addServiceSynchronizer($id, $definition);
```
Comment (sstok) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: The `$synchronizers` variable can also be removed as its not really used anymore now.

Comment (dosten) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Done!

**Comment by sstok** [comment]
LGTM :+1:

File: src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php [hunk 2]
```diff
@@ -148,7 +148,7 @@ private function isInlineableDefinition(ContainerBuilder $container, $id, Defini
             return false;
         }
 
-        if (count($ids) > 1 && $definition->getFactoryService(false)) {
+        if (count($ids) > 1 && $definition->getFactory()) {
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php]: this should be removed entirely instead. The case of factories is already covered by the previous condition

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/xml/services20.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
   <services>
-    <service id="request" class="Request" synthetic="true" synchronized="true"/>
+    <service id="request" class="Request" synthetic="true"/>
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/xml/services20.xml]: why not removing the file entirely like for other formats ?

**Comment by stof** [comment]
this is likely to conflict with #14127

File: src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php (hunk 2 above)
Comment (dosten) [inline src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php]: Done

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/xml/services20.xml (hunk 3 above)
Comment (dosten) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/xml/services20.xml]: You are right, done!

**Comment by fabpot** [comment]
Thank you @dosten.

File: src/Symfony/Component/DependencyInjection/Container.php [hunk 4]
```diff
@@ -198,11 +196,9 @@ public function set($id, $service, $scope = self::SCOPE_CONTAINER)
         $id = strtolower($id);
 
         if ('service_container' === $id) {
-            // BC: 'service_container' is no longer a self-reference but always
-            // $this, so ignore this call.
-            // @todo Throw InvalidArgumentException in next major release.
-            return;
+            throw new InvalidArgumentException('You cannot set service "service_container".');
```
Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Container.php]: this and also the special logic in `has` looks wrong to me. the container should not know anything about the `service_container` anymore and just treat it like any other service.

Comment (Tobion) [inline src/Symfony/Component/DependencyInjection/Container.php]: nvm, it's purpose is to not be able to overwrite the special service which is ok

Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Container.php]: Hm, do we actually need this kind of special service at all or could we get rid of it?

Comment (hacfi) [inline src/Symfony/Component/DependencyInjection/Container.php]: @xabbuh I think we need it to inject the entire container into a service (not recommended I know but a lot of developers do)

Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Container.php]: @hacfi That's true.

**Comment by hacfi** [comment]
@dosten very nice work!

**Pull request closed without merging**